src/runtime/constructorbinding.cs

```diff
@@ -19,18 +19,20 @@ namespace Python.Runtime
     /// and creating the BoundContructor object which contains ContructorInfo object.
     /// 3) In tp_call, if ctorInfo is not null, ctorBinder.InvokeRaw() is called.
     /// </remarks>
+    [Serializable]
     internal class ConstructorBinding : ExtensionType
     {
         private Type type; // The managed Type being wrapped in a ClassObject
         private IntPtr pyTypeHndl; // The python type tells GetInstHandle which Type to create.
         private ConstructorBinder ctorBinder;
+
+        [NonSerialized]
         private IntPtr repr;
 
         public ConstructorBinding(Type type, IntPtr pyTypeHndl, ConstructorBinder ctorBinder)
         {
             this.type = type;
-            Runtime.XIncref(pyTypeHndl);
-            this.pyTypeHndl = pyTypeHndl;
+            this.pyTypeHndl = pyTypeHndl; // steal a type reference
```

Comment:
As long as TypeManager.cache exists, if there's no bug in program, it won't be 0, also it wouldn't be access after `TypeManager.RemoveTypes` called, the `pyHandle`(attention: not pyTypeHndl) already had been deduct to a normal object.
I remove the XIncref line simply just because the original code didn't make the reference count balance(`tp_dealloc` didn't decrease it).
If XIncref here, it must XDecref in tp_clear and tp_dealloc, otherwise not.